node/connection_base.js

```diff
@@ -113,6 +114,13 @@ TChannelConnectionBase.prototype.onTimeoutCheck = function onTimeoutCheck() {
     }
     if (self.lastTimeoutTime) {
         self.timedOutEvent.emit(self);
+    } else if (!self.remoteName &&
+        self.timers.now() - self.startTime > self.channel.requestDefaults.timeout) {
+        self.timedOutEvent.emit(self, errors.TimeoutError({
```

Comment:
you should just emit an error here instead of a timedOutEvent.

If the connection is not initialized its broken. We already listen on the `close` event in node/request.js so there should be no extra events needed there